### PR TITLE
affine: init at 0.13.1

### PR DESCRIPTION
**Pull request by ri-char**
## Description of changes

Add [AFFiNE](https://affine.pro/), an open-source, all-in-one workspace and an operating system for all the building blocks that assemble your knowledge base and much more -- wiki, knowledge management, presentation and digital assets.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/af/affine/package.nix [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      desktopName = "AFFiNE";
+      exec = "affine %U";
+      terminal = false;
+      icon = "affine";
+      startupWMClass = "affine";
+      categories = ["Utility"];
+    })
+  ];
+  meta = with lib; {
+    description = ''
```
Comment (SuperSandro2000) [inline pkgs/by-name/af/affine/package.nix:66]: Multiline descriptions should go to longDescription

File: pkgs/by-name/af/affine/package.nix [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    url = "https://github.com/toeverything/AFFiNE/releases/download/v${finalAttrs.version}/affine-${finalAttrs.version}-stable-linux-x64.zip";
+    hash = "sha256-2Du5g/I82iTr8Bwb+qkLzyfbk1OrOlXqx6FHImVoAoE=";
+  };
+  nativeBuildInputs = [
+    copyDesktopItems
+  ];
+  sourceRoot = ".";
+  unpackPhase = ''
+    mkdir $out
+    ${unzip}/bin/unzip $src -d $out
```
Comment (SuperSandro2000) [inline pkgs/by-name/af/affine/package.nix:44]:
```suggestion
    unzip $src -d $out
```
unzip should go to nativeBuildInputs

File: pkgs/by-name/af/affine/package.nix [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    hash = "sha256-smZ5W7fy3TK3bvjwV4i71j2lVmKSZcyhMhcWfPxNnN4=";
+  };
+  script = writeText "affine" ''
+    #!${bash}/bin/bash
+    set -e
+    export ELECTRON_IS_DEV=0
+    export ELECTRON_FORCE_IS_PACKAGED=true
+    export NODE_ENV=production
+    cd @out@
+    if [[ $EUID -ne 0 ]] || [[ $ELECTRON_RUN_AS_NODE ]]; then
```
Comment (SuperSandro2000) [inline pkgs/by-name/af/affine/package.nix:25]: This should be done mostly with makeWrapper.

File: pkgs/by-name/af/affine/package.nix [hunk 4]
```diff
@@ -0,0 +1,79 @@
+{ lib
+, writeText
+, fetchurl
+, stdenvNoCC
+, copyDesktopItems
+, makeDesktopItem
+, unzip
+, bash
+, electron
+, commandLineArgs ? ""
```
Comment (SuperSandro2000) [inline pkgs/by-name/af/affine/package.nix:10]: Why do we need to rebuild the entire package just to change the arguments to the wrapper?

Comment (ri-char) [reply pkgs/by-name/af/affine/package.nix:11]: Electron applications usually need this. It allows user to pass extra arguments to electron. For example, `--enable-wayland-ime` is needed if someone wants to use input method in wayland.

**Comment by happysalada** [comment]
Hey thank you for this, this is very nice ! Do you have plans for makign a service, is there anything you need help with ?

**Comment by ri-char** [comment]
> Hey thank you for this, this is very nice ! Do you have plans for makign a service, is there anything you need help with ?

Hi, thanks for your kind words! I currently don't have plans to add this, but feel free to submit a PR if you'd like to contribute.

**Pull request merged**